### ISSUE TITLE
do not require a coloured Unicode terminal for opam-ci

### ISSUE THREAD

**Issue by avsm** [ocaml/obi#6]
The current dashboard display requires an ANSI and Unicode tty due to the use of colours.  This is not optimal for several reasons:
- it makes it difficult to use with screen readers
- file redirection is messy
- it also fails to work on Windows due to Consolas not supporting the right format (via @dra27)

We should fallback to a more verbose rendering format in these situations instead of blasting escape codes to the screen unconditionally.

**Comment by samoht** [comment]
Also it doesn't help with copy/pasting code. Ideally the success and failure symbols should just be different, even if they have different colors. I advise to pick a consistent style, e.g. all successes are in circles and errors in square boxes or something like that.

**Comment by avsm** [comment]
Agreed, but the difficulty here is just finding a good Unicode symbol set that [also renders on Windows](http://www.fileformat.info/info/unicode/font/consolas/grid.htm).  Suggestions very welcome -- the relevant code to be factored is in [src/opam-ci/user.ml#L24](https://github.com/ocaml/obi/blob/master/src/opam-ci/user.ml#L24)

**Comment by trefis** [comment]
To go a bit further: I think a `--ascii` option is desirable (even though it makes the output less compact).

**Comment by samoht** [comment]
One idea would also be to use a simple table. 

So instead of:
```
                     alcotest-async          0.8.2  ③ ④ ⑤ ⑥ ⑦ ⑧   🄳 🄰 🅄 🄵 🅂   ⓧ ⓐ ⓟ   ∬ ﬂ ⚐
                            datakit         0.12.0  ③ ④ ⑤ ⑥ ⑦ ⑧   🄳 🄰 🅄 🄵 🅂   ⓧ ⓐ ⓟ   ∬ ﬂ ⚐
              datakit-bridge-github         0.12.0  ③ ④ ⑤ ⑥ ⑦ ⑧   🄳 🄰 🅄 🄵 🅂   ⓧ ⓐ ⓟ   ∬ ﬂ ⚐
```

Have something like that:

```
                               ③ ④ ⑤ ⑥ ⑦ ⑧   🄳 🄰 🅄 🄵 🅂   ⓧ ⓐ ⓟ   ∬ ﬂ ⚐
alcotest-async        0.8.2    ✓  ✓  ✓  ✓ ✗  ✗   ✓  ✗  ✗  ✗  ✓   ✓  ✓  ✓   ✓ ✗ ✗
datakit               0.12.0   ✓  ✓  ✓  ✓ ✗  ✗   ✓  ✗  ✗  ✗  ✓   ✓  ✓  ✓   ✓ ✗ ✗
datakit-bridge-github 0.12.0   ✗  ✓  ✓  ✗ ✗  ✗   ✓  ✗  ✗  ✗  ✓   ✗  ✓  ✓   ✓ ✗ ✓
```

**Comment by samoht** [comment]
Also it would be nice if the columns to display should be customisable, for instance using a config file. Also maybe it's not worth showing the columns where everything is fine :-)

**Comment by dbuenzli** [comment]
Since you are already using `Fmt`, you might be interested in using [`Fmt_tty`](http://erratique.ch/software/fmt/doc/Fmt_tty.html) which has support for configuring [conditional UTF-8 output](http://erratique.ch/software/fmt/doc/Fmt.html#utf8_cond) and ANSI coloring in some kind of standard way. 

Besides [`Fmt_cli`](http://erratique.ch/software/fmt/doc/Fmt_cli.html) has a cmdliner fragment to handle an explicit `--color` argument, possibly backed by a custom environment variable. 

Here's a [sample usage](https://github.com/dbuenzli/carcass/blob/fc4613764fb36568eb2a599b2e56af6f069148bc/src-bin/cli.ml#L97-L114) for the coloring bits.
